cellfinder_napari/detect_utils.py

```diff
@@ -0,0 +1,186 @@
+from dataclasses import dataclass, asdict
+from pathlib import Path
+
+import napari
+from cellfinder_core.main import main as cellfinder_run
+from imlib.cells.cells import Cell
+from napari.qt.threading import thread_worker
+
+from cellfinder_napari.utils import cells_to_array
+
+
+@dataclass
+class InputContainer:
+    """Base data class to group inputs into a container, with persistent defaults"""
+
+    persistent_defaults = (
+        {}
+    )  # some elements need to be remembered without having to create a new instance, for reset button
+
+    @classmethod
+    def numerical_widget(
+        cls, key, custom_label=None, step=0.1, min=None, max=None
+    ) -> dict:
+        """Provide info to populate a numerical widget with the class's persistent defaults."""
+        # TODO: maybe some assertions like either both or neither min/max==None and cls.defaults[key] is numerical.
+        if custom_label is None:
+            label = key.replace("_", " ").capitalize()
+        else:
+            label = custom_label
```

Comment:
Are we being consistent with capitalisation? i.e. capital just for first word rather than all

```suggestion
     label = key.replace("_", " ").capitalize() if custom_label is None else custom_label
```

Reply:
Yes re capitalisation (at least that was the principle before this PR IIUC, and this PR should not change the UI.
I've added your suggestion here - thanks.